### DISCUSSION TITLE
how to use node-fetch with AdonisJs v5?

### DISCUSSION THREAD

**Discussion by lenovo1996** [adonisjs/core#3956] [Help] [answered]
Hello AdonisJs!
I am a newbie in Nodejs (typescript) and I very love this framework.
currently, I make a project with AdonisJs but got some problems.

When I use node-fetch version 32.10 to call another API from Task. I got this error.
I try to search in stackoverflow to resolve this problem. but failed.
``` Must use import to load ES Module: api/node_modules/node-fetch/src/index.js
require() of ES modules is not supported.
require() of api/node_modules/node-fetch/src/index.js from api/app/Containers/v1/Crawler/Task/CallApiTask.ts is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from api/node_modules/node-fetch/package.json.
```

please help me resolve this. thanks

this is my code:

```
import BaseTask from "Porto/Task/BaseTask";
import fetch from 'node-fetch';

export default class CallApiTask extends BaseTask {
    constructor() {
        super();
    }

    async handle(url) {
        try {
            const response = await fetch(url);
            return await response.json();
        } catch (e) {
            return {};
        }
    }
}
```

**Comment by lenovo1996** [comment]
I try with some other packages like `got` but still got the same error.

**Comment by thetutlage** [comment, marked as answer]
As the error message tells `require() of ES modules is not supported.`. The version of the packages you are using are ESM only, so they cannot work with CommonJS modules. You need to downgrade `got` to a version that supports CommonJS

**Reply by lenovo1996** [reply]
thank you so much. 
When I downgrade to got v11 it ran ok.
